lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/PersistableProcessor.java

```diff
@@ -0,0 +1,189 @@
+package org.enso.interpreter.dsl;
+
+import java.io.IOException;
+import java.util.Set;
+
+import javax.annotation.processing.AbstractProcessor;
+import javax.annotation.processing.Processor;
+import javax.annotation.processing.RoundEnvironment;
+import javax.annotation.processing.SupportedAnnotationTypes;
+import javax.lang.model.SourceVersion;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.Modifier;
+import javax.lang.model.element.PackageElement;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.type.MirroredTypeException;
+import javax.tools.Diagnostic.Kind;
+
+import org.openide.util.lookup.ServiceProvider;
+
+@SupportedAnnotationTypes({
+  "org.enso.interpreter.dsl.Persistable",
+  "org.enso.interpreter.dsl.Persistable.Group"
+})
+@ServiceProvider(service=Processor.class)
+public class PersistableProcessor extends AbstractProcessor {
```

Comment:
Missing some documentation on what does this processor does (simple example of input/output)

Reply:
Done in [b22f5eb](https://github.com/enso-org/enso/pull/8207/commits/b22f5eb29f7ca9117999326e7a4429847df80aef)